File: app/src/app/_domain/chamado.ts
import {JfFiles} from './jfFiles';
import {Comentario} from './comentario';

export class Chamado {
    id: number;
    titulo: string;
    versao: string;
    usuarioID: number;
    atribuicaoID: number;
    ocorrenciaID: number;
    contatoID: number;
    contatoNome: string;
    clienteID: number;
    datAbertura: Date;
    datFechamento: Date;
    detalhamento: string;
    sistemas: string;
    palavraChave: string;
    prioridade = 1;
    situacao = 2;
    atendimento: string;
    solucao: string;
    cliente: string;
    anexos: JfFiles[] = [];
    comentarios: Comentario[] = [];
}

File: api/src/main/kotlin/br/com/oxent/sfarma/model/entity/Chamado.kt
package br.com.oxent.sfarma.model.entity

import java.time.LocalDateTime

class Chamado(
        var id: Int?,
        val versao: String?,
        val titulo: String?,
        val usuarioID: Int,
        var atribuicaoID: Int?,
        val ocorrenciaID: Int,
        val contatoID: Int?,
        val clienteID: Int?,
        val datAbertura: LocalDateTime,
        var datFechamento: LocalDateTime?,
        val detalhamento: String,
        val sistemas: String?,
        val palavaChave: String? = null,
        val prioridade: Int,
        val atendimento: String?,
        val situacao: Int? = 0,
        val solucao: String? = null,
        var anexos: List<JFFile> = mutableListOf(),
        var comentarios: List<Comentario> = mutableListOf()
)